rootfs/scheduler/__init__.py

```diff
@@ -390,6 +397,13 @@ def deploy(self, namespace, name, image, command, **kwargs):
         # New release is live and kicking. Clean up old release
         if old_rc:
             self._delete_rc(namespace, old_rc["metadata"]["name"])
+            # remove secret that contains env vars for the release
+            secret_name = "{}-{}-env".format(namespace, old_rc['metadata']['labels']['version'])
+            try:
+                self._get_secret(namespace, secret_name)
+                self._delete_secret(namespace, secret_name)
```

Comment:
I can fire off the delete straight up but felt it was more appropriate to hit the k8s API cache before trying to delete from it, probably an overkill but doesn't hurt either